components/org.wso2.carbon.identity.media.endpoint/src/main/java/org/wso2/carbon/identity/media/endpoint/common/Util.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.media.endpoint.common;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.log4j.MDC;
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.core.ServiceURLBuilder;
+import org.wso2.carbon.identity.core.URLBuilderException;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.media.core.StorageSystemManager;
+import org.wso2.carbon.identity.media.endpoint.Error;
+import org.wso2.carbon.identity.media.endpoint.exception.MediaEndpointException;
+
+import java.net.URI;
+import java.util.UUID;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.core.util.IdentityCoreConstants.TENANT_NAME_FROM_CONTEXT;
+import static org.wso2.carbon.identity.media.endpoint.common.MediaServiceConstants.CORRELATION_ID_MDC;
+import static org.wso2.carbon.identity.media.endpoint.common.MediaServiceConstants.MEDIA_API_PATH_COMPONENT;
+import static org.wso2.carbon.identity.media.endpoint.common.MediaServiceConstants.TENANT_CONTEXT_PATH_COMPONENT;
+import static org.wso2.carbon.identity.media.endpoint.common.MediaServiceConstants.V1_API_PATH_COMPONENT;
+
+/**
+ * This class includes the utility methods for media endpoint.
+ */
+public class Util {
```

Comment:
let's make the constructor private

Reply:
fixed with: b04c81c84c80a9eef1a43eb56605b6224a3cb7cc